src/infra/cache/tmpfs.rs

```diff
@@ -165,16 +165,16 @@ mod tests {
     fn create_read_file() {
         let path = "/tmp/test.txt";
         let _ = write_file(path, b"hello world").unwrap();
-        let data = read_file(path).unwrap();
-        assert_eq!(data, b"hello world");
+        assert_eq!(read_file(path).unwrap(), b"hello world");
         remove_file(path).unwrap();
     }
 
     #[test]
     fn create_read_directory() {
-        assert_eq!(true, create_dir_all("/tmp/test_dir/abc/").is_ok());
-        assert_eq!(true, create_dir("/tmp/test_dir/abc/cde/").is_ok());
-        assert_eq!(true, remove_dir("/tmp/test_dir/abc/cde/").is_ok());
-        assert_eq!(true, remove_dir_all("/tmp/test_dir/").is_ok());
+        assert!(create_dir("/tmp/test_dir/abc/cde/").is_err());
+        create_dir_all("/tmp/test_dir/abc/").unwrap();
+        create_dir("/tmp/test_dir/abc/cde/").unwrap();
+        remove_dir("/tmp/test_dir/abc/cde/").unwrap();
+        remove_dir_all("/tmp/test_dir/").unwrap();
```

Comment:
you removed assert maybe cause panic if the function not work.

Reply:
[`Result::unwrap`](https://doc.rust-lang.org/std/result/enum.Result.html#method.unwrap) does panic on `Err` value.

We shouldn't use `unwrap` in production code or documentation, but unwrapping in tests is totally acceptable.

```rust
pub fn ensure_oddity(n: u8) -> Result<(), String> {
    if n % 2 == 0 {
        Err(format!("{n} is even"))
    } else {
        Ok(())
    }
}

#[test]
fn test_odd() {
    assert!(ensure_oddity(13).is_ok());
    ensure_oddity(7).unwrap();
}

#[test]
#[should_panic]
fn test_even() {
    ensure_oddity(8).unwrap();
}

#[test]
#[should_panic]
fn test_zero() {
    assert!(ensure_oddity(0).is_ok());
}
```

[Playground link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=ba0b934b1b5bfaa40a7654af6d453b43)

Reply:
You added the unwrap.